### DISCUSSION TITLE
Use decimal value type for tablet input area settings

### DISCUSSION THREAD

**Discussion by Methos-** [ppy/osu#13188] [Ideas]
**Describe the bug:** Messed around with the new update, had a lot of fun using the built in tablet support. I had no ability to use decimals so I decided to pass them through the configuration file (input.json) instead. Worked perfectly in-game, just that it seems it cause the tablet input options screen to bug out a bit. 

**Content of the input.json file:**
`{"InputHandlers":[{"$type":"osu.Framework.Input.Handlers.Keyboard.KeyboardHandler, osu.Framework","Description":"Keyboard","IsActive":true,"Enabled":true},{"$type":"osu.Framework.Input.Handlers.Tablet.OpenTabletDriverHandler, osu.Framework","IsActive":true,"AreaOffset":{"x":58.21,"y":43.55},"AreaSize":{"x":75.3,"y":42.36},"Rotation":352.5,"Tablet":{"Name":"Wacom CTH-480","Size":{"x":152.0,"y":95.0}},"Description":"OpenTabletDriver","Enabled":true},{"$type":"osu.Framework.Input.Handlers.Joystick.JoystickHandler, osu.Framework","Description":"Joystick / Gamepad","IsActive":true,"Enabled":true},{"$type":"osu.Framework.Input.Handlers.Midi.MidiHandler, osu.Framework","Description":"MIDI","IsActive":true,"Enabled":true},{"$type":"osu.Framework.Platform.Windows.WindowsMouseHandler, osu.Framework","IsActive":true,"UseRelativeMode":true,"Sensitivity":1.0,"Description":"Mouse","Enabled":true}]}`

**Screenshots or videos showing encountered issue:** 
![osu_2021-04-02_00-39-56](https://user-images.githubusercontent.com/6557053/113394024-65398c80-934c-11eb-9aff-8d477af2b2ce.jpg)

**osu!lazer version:** 2021.331.0

**Logs:** 
[network.log](https://github.com/ppy/osu/files/6248387/network.log)
[performance.log](https://github.com/ppy/osu/files/6248388/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6248389/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6248390/updater.log)
[database.log](https://github.com/ppy/osu/files/6248391/database.log)

**Comment by peppy** [comment]
This issue should probably be reworded to request decimal input support at the settings. We don't support manually editing configuration files.

**Reply by taep96** [reply]
Can you provide an ETA of when this is going to be implemented?

**Reply by bdach** [reply]
@LemonJuicer96 no. please do not ask.

**Reply by Samir2zk** [reply]
Is there any actual way of putting the decimals or when you doing it bugs?

**Reply by Methos-** [reply]
> Is there any actual way of putting the decimals or when you doing it bugs?

input.json file, after that you can't touch the values on the tablet options otherwise it will reset.

**Comment by vatei** [comment]
**Edit: new controls now allows this feature to exist, would be nice as a followup**

Sorry for the bump, but decimal input (and manual entry as mentionned here https://github.com/ppy/osu/discussions/25134#discussioncomment-7287929) are likely important, as new people are going to use the new client for the first time soon. They would probably want to use their existing configuration from OpenTabletDriver.

Thanks in advance and good job for all the work done so far! :)

**Comment by SupDos** [comment]
I wanted to give this a try as I was wanting to use the internal OTD to make it easier to use my tablet on other devices without external OTD installed, but currently am unable to as my tablet area has decimals and I can definitely tell something feels off if I use whole numbers instead. I thought it would be as easy as changing the precision on the *Width* and *Height* controls to have as much precision as external OTD allows, but I noticed something weird.

For some reason, switching the values manually and pressing the "Confirm to aspect ratio" button gave different results for the values in the controls. 

https://github.com/user-attachments/assets/39567945-01dd-4003-bb8a-b396dd1fd99e


(notice how Height changes after pressing the button)

> for reference, these are the values I get in OTD for all 3 examples, and what I'm considering are the "correct" values:
> <img width="1520" height="70" alt="image" src="https://github.com/user-attachments/assets/696b2700-1f45-4882-a9e5-91b4e00cdce9" />



As you can see, the correct values are the ones I get *after* pressing the conform button.

After spending a good chunk of time trying to figure out why both of these were giving different values, I realized that editing the value manually and pressing the button gives different precision of aspect ratio, which ends up causing the slight difference in value. Manually editing causes the aspect ratio to be clamped due to its precision being *0.01f*:
https://github.com/ppy/osu/blob/95648a3d27a0c9c4e544c445f9baba4a8c5b8948/osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs#L67-L72
https://github.com/ppy/osu/blob/95648a3d27a0c9c4e544c445f9baba4a8c5b8948/osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs#L308-L309

<img width="237" height="69" alt="image" src="https://github.com/user-attachments/assets/58ce051b-e64c-42a4-86bd-bb2f520d2ff1" />


while pressing the button seemingly skips this and gives an unclamped aspect ratio value by doing its own calculation for it (i think?):
https://github.com/ppy/osu/blob/95648a3d27a0c9c4e544c445f9baba4a8c5b8948/osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs#L130-L131
https://github.com/ppy/osu/blob/95648a3d27a0c9c4e544c445f9baba4a8c5b8948/osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs#L139
https://github.com/ppy/osu/blob/95648a3d27a0c9c4e544c445f9baba4a8c5b8948/osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs#L321-L325

<img width="271" height="66" alt="image" src="https://github.com/user-attachments/assets/162b29bf-60e4-474b-bfdb-9b67daff6565" />

Increasing the aspectRatio precision to something like *0.00001f* fixes the issue, but this definitely seems like the wrong way to fix it so I'm leaving this here in the hopes that someone more intelligent than me can help figure it out...
